tsuru_dashboard/admin/views.py

```diff
@@ -182,22 +181,21 @@ def nodes_by_pool(self, pool):
             return pools
 
         data = response.json()
-        nodes = data.get("nodes", [])
+        node_list = data.get("nodes", [])
+        nodes = map(lambda n: Node(n), node_list)
+        nodes = filter(lambda n: n.pool() == pool, nodes)
 
         url = "{}/docker/node/{}/containers"
 
         rs = []
         for node in nodes:
-            if Node(node).pool() == pool:
-                u = url.format(settings.TSURU_HOST, node["Address"])
-                rs.append(grequests.get(u, headers=self.authorization))
+            u = url.format(settings.TSURU_HOST, node.address())
+            rs.append(grequests.get(u, headers=self.authorization))
 
         units = grequests.map(rs)
 
-        for node in nodes:
-            node = Node(node, units)
-            if node.pool() != pool:
-                continue
+        for i, node in enumerate(nodes):
+            node.load_units(units[i])
```

Comment:
Is it guaranteed that `grequests.map` return will preserve the order from the `rs` object? I think the old code was so ugly because we were not sure.

Reply:
From the source code, seems like `grequests.map` does preserve the order: https://github.com/kennethreitz/grequests/blob/master/grequests.py#L103